lang/java/reef-bridge-client/src/main/java/org/apache/reef/bridge/client/YarnJobSubmissionClient.java

```diff
@@ -207,7 +206,65 @@ private void launch(final YarnSubmissionFromCS yarnSubmission) throws IOExceptio
       } catch (InjectionException ie) {
         throw new RuntimeException("Unable to submit job due to " + ie);
       }
+      writeDriverHttpEndPoint(yarnSubmission.getDriverFolder(),
+          submissionHelper.getStringApplicationId(), jobFolderOnDFS.getPath());
+    }
+  }
+
+  /**
+   * We leave a file behind in job submission directory so that clr client can figure out
+   * the applicationId and yarn rest endpoint.
+   * @param driverFolder
+   * @param applicationId
+   * @throws IOException
+   */
+  private void writeDriverHttpEndPoint(final File driverFolder,
+                                       final String applicationId,
+                                       final Path dfsPath) throws  IOException {
+    final FileSystem fs = FileSystem.get(yarnConfiguration);
+    final Path httpEndpointPath = new Path(dfsPath, fileNames.getDriverHttpEndpoint());
+
+    String trackingUri = null;
+    for (int i = 0; i < 60; i++) {
+      try {
+        LOG.log(Level.INFO, "Attempt " + i + " reading " + httpEndpointPath.toString());
+        if (fs.exists(httpEndpointPath)) {
+          FSDataInputStream input = fs.open(httpEndpointPath);
+          BufferedReader reader = new BufferedReader(new InputStreamReader(input));
```

Comment:
In REEF-768 we want to eliminate high-priority warnings from Findbugs; this line will (likely) produce a new warning, `DM_DEFAULT_ENCODING: Reliance on default encoding` which complains about not setting encoding explicitly. Adding "UTF-8" as second parameter for `InputStreamReader` constructor should fix it.